kernel/src/scan/mod.rs

```diff
@@ -158,31 +166,32 @@ impl Scan {
     /// from the deletion vector if it was present. See the documentation for [`ScanResult`] for
     /// more details.
     pub fn execute(&self, engine_interface: &dyn EngineInterface) -> DeltaResult<Vec<ScanResult>> {
-        let parquet_handler = engine_interface.get_parquet_handler();
-
         let partition_columns = &self.snapshot.metadata().partition_columns;
-        let read_schema = Arc::new(StructType::new(
-            self.schema()
-                .fields()
-                .filter(|f| !partition_columns.contains(f.name()))
-                .cloned()
-                .collect::<Vec<_>>(),
-        ));
-
-        let mut partition_fields = partition_columns
-            .iter()
-            .map(|column| {
-                self.schema()
-                    .field(column)
-                    .ok_or(Error::generic("Unexpected partition column"))
-            })
-            .collect::<DeltaResult<Vec<_>>>()?;
-        partition_fields.reverse();
-
-        let select_fields = read_schema
+        let mut read_fields = vec![];
```

Comment:
Should we use `with_capacity(self.schema.len())`, since there will be 0-4 partition cols at most?